crypto/fipsmodule/rand/entropy/entropy_source_tests.cc

```diff
@@ -0,0 +1,40 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0 OR ISC
+
+#include <gtest/gtest.h>
+
+#include "internal.h"
+
+#define MAX_MULTIPLE_FROM_RNG (16)
+
+// In the future this test can be improved by being able to predict whether the
+// test is running on hardware that we expect to support RNDR. This will require
+// amending the CI with such information.
+// For now, simply ensure we exercise all code-paths in the
+// CRYPTO_rndr_multiple8 implementation.
+TEST(EntropySupport, Aarch64) {
+  uint8_t buf[MAX_MULTIPLE_FROM_RNG*8] = { 0 } ;
+
+#if !defined(OPENSSL_AARCH64)
+  ASSERT_FALSE(have_hw_rng_aarch64_for_testing());
+  ASSERT_FALSE(rndr_multiple8(buf, 0));
+  ASSERT_FALSE(rndr_multiple8(buf, 8));
+#else
+  if (have_hw_rng_aarch64_for_testing() != 1) {
+    GTEST_SKIP() << "Compiled for Arm64, but Aarch64 hw rng is not available in run-time";
+  }
+
+  // Extracting 0 bytes is never supported.
+  ASSERT_FALSE(rndr_multiple8(buf, 0));
+
+  // Multiples of 8 allowed.
+  for (size_t i = 8; i < MAX_MULTIPLE_FROM_RNG; i += 8) {
+    ASSERT_TRUE(rndr_multiple8(buf, i));
```

Comment:
Would you like to do a simple check that it's a new value every time to make sure it's not stuck? Or there are other tests in the code doing that whenever it's available?

Reply:
The next part of the new randomness generation implementation is to port over existing tests and also improve testing. I don't think there is a test that checks whether the entropy sources can be stuck on any values.
But I will add some basic tests to the existing test suite that covers the obvious "okay, this is quite broken"-case.